docs/getting_started.rst

```diff
@@ -399,12 +399,16 @@ Or to check the formatting without writing changes, run:
   yarn run lint-frontend
   yarn run fmt-backend:check
 
-To have code automatically formatted and checked for linting upon commit, you may use `pre-commit <http://pre-commit.com/>`__ this can help ensure clean, consistent code, and will prevent automated build checks due to linting errors. The pip package should already be installed from the Python dev dependency installation, but you need to install the git hooks using this command:
+`pre-commit <http://pre-commit.com/>`__ is used to apply a full set of checks and formatting automatically each time that ``git commit`` runs. If there are errors, the Git commit is aborted and you are asked to fix the error and run ``git commit`` again.
+
+Pre-commit is already installed as a development dependency, but you also need to enable it:
 
 .. code-block:: bash
 
   pre-commit install
 
+.. note:: Pre-commit can have issues running from alternative Git clients like GitUp. If you encounter problems while committing changes, run ``pre-commit uninstall`` to disable pre-commit.
```

Comment:
```suggestion
.. note:: Installing pre-commit will prevent you from checking in-progress code into your local working branch, which might not be desired. Pre-commit can have issues running from GUI Git clients because it needs to be run within your Python virtual environment. If you encounter problems while committing changes, run ``pre-commit uninstall`` to disable pre-commit.
```

Reply:
I thought we had resolved that pre-commit itself does not prevent this, but it's the usage of a git client that doesn't support virtual environments launched from the git hook?

Alternatives are: `git commit --no-verify` and installing the hook as a pre-push hook? Shouldn't that rather be documented instead of passing false information? Pre-commit is a wonderful and powerful tool, I feel like it's unnecessarily and unfairly discouraged in this manner.

Reply:
okay, updated with additional information so people can make informed choices